sklearn/svm/base.py

```diff
@@ -482,8 +482,8 @@ def _validate_for_predict(self, X):
     @property
     def coef_(self):
         if self.kernel != 'linear':
-            raise ValueError('coef_ is only available when using a '
-                             'linear kernel')
+            raise AttributeError('coef_ is only available when using a '
```

Comment:
I thought I had suggested `ParamDependentAttributeError` (which makes sense in terms of being a ValueError-dependent AttributeError). Still, it seems analogous to too much red tape.